homeassistant/components/device_tracker/bluetooth_tracker.py

```diff
@@ -0,0 +1,90 @@
+"""Tracking for bluetooth devices."""
+import logging
+import os
+from datetime import timedelta
+
+from homeassistant.helpers.event import track_point_in_utc_time
+from homeassistant.components.device_tracker import (
+    YAML_DEVICES,
+    CONF_TRACK_NEW,
+    DEFAULT_CONF_TRACK_NEW,
+    CONF_SCAN_INTERVAL,
+    DEFAULT_SCAN_INTERVAL,
+)
+from homeassistant.config import load_yaml_config_file
+import homeassistant.util as util
+import homeassistant.util.dt as dt_util
+
+_LOGGER = logging.getLogger(__name__)
+
+REQUIREMENTS = ['pybluez==0.22']
+
+
+def setup_scanner(hass, config, see):
+    """Setup the Bluetooth Scanner."""
+    import bluetooth
+
+    def see_device(device, see):
+        """Mark a device as seen."""
+        see(dev_id='bt_' + device[1], mac=device[0], host_name=device[1])
+
+    def discover_devices():
+        """Discover bluetooth devices."""
+        result = bluetooth.discover_devices(duration=8,
+                                            lookup_names=True,
+                                            flush_cache=True,
+                                            lookup_class=False)
+        _LOGGER.info("Bluetooth devices discovered = " + str(len(result)))
+        return result
+
+    yaml_path = hass.config.path(YAML_DEVICES)
+    devs_to_track = []
+    devs_donot_track = []
+
+    # load all known devices
+    if os.path.isfile(yaml_path):
+        for dev_id, device in load_yaml_config_file(yaml_path).items():
+            # check if device is a valid bluetooth device
+            if dev_id.startswith('bt_') and device['mac']:
+                if device['track']:
+                    devs_to_track.append(device['mac'])
+                else:
+                    devs_donot_track.append(device['mac'])
+
+    # if track new devices is true discover new devices
+    # on startup.
+    track_new = util.convert(config.get(CONF_TRACK_NEW), bool,
+                             DEFAULT_CONF_TRACK_NEW)
```

Comment:
`DEFAULT_CONF_TRACK_NEW` should probably be `True` if there are no devices found to track? So replace it with:
```python
len(devs_to_track) == 0
```